lib/spack/spack/binary_distribution.py

```diff
@@ -534,7 +534,22 @@ def relocate_package(workdir, spec, allow_root):
                            newprefix=new_prefix)
     # If the binary files in the package were not edited to use
     # relative RPATHs, then the RPATHs need to be relocated
-    if not rel:
+    if rel:
+        if old_path != new_path:
+            files_to_relocate = list(filter(
```

Comment:
This is a bit much to unpack. I think the filter function ought to be defined separately. 

Also, should we report those members of `relocate_binaries` that are not in fact relocatable?

Reply:
We get a lot of false positives for inability to relocate, because lots of things have paths hardcoded in their dwarf info or elsewhere that do not affect correctness. That's why I switched to a single warning per-package.